src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -314,6 +320,15 @@ public CASFileCache(
     this.delegateSkipLoad = delegateSkipLoad;
     this.directoriesIndexDbName = directoriesIndexDbName;
 
+    if (configs.getWorker().getStorages().get(0).isPublishTtlMetric()) {
```

Comment:
We really should be taking a config object as a param if you're going to use it this way. We don't know that we're the 0-indexed config for that tree, and this makes things lots less testable (without a config instance setup in every test class).

Reply:
fixed to not use 0 indexing.  Now the CAS gets appropriate settings on construction